pkg/resource/enqueue_handlers.go

```diff
@@ -71,3 +72,41 @@ func addPropagated(obj runtime.Object, queue adder) {
 		queue.Add(reconcile.Request{NamespacedName: nn})
 	}
 }
+
+// EnqueueRequestForProviderConfig enqueues a reconcile.Request for a referenced
+// ProviderConfig.
+type EnqueueRequestForProviderConfig struct{}
+
+// Create adds a NamespacedName for the supplied CreateEvent if its Object is a
+// ProviderConfigReferencer.
+func (e *EnqueueRequestForProviderConfig) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	addProviderConfig(evt.Object, q)
+}
+
+// Update adds a NamespacedName for the supplied UpdateEvent if its Objects are
+// a ProviderConfigReferencer.
+func (e *EnqueueRequestForProviderConfig) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	addProviderConfig(evt.ObjectOld, q)
+	addProviderConfig(evt.ObjectNew, q)
+}
+
+// Delete adds a NamespacedName for the supplied DeleteEvent if its Object is a
+// ProviderConfigReferencer.
+func (e *EnqueueRequestForProviderConfig) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	addProviderConfig(evt.Object, q)
+}
+
+// Generic adds a NamespacedName for the supplied GenericEvent if its Object is
+// a ProviderConfigReferencer.
+func (e *EnqueueRequestForProviderConfig) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	addProviderConfig(evt.Object, q)
+}
+
+func addProviderConfig(obj runtime.Object, queue adder) {
+	pcr, ok := obj.(RequiredProviderConfigReferencer)
```

Comment:
I guess you're trying to make sure a reconcile request is accepted for `ProviderConfig` via events on `ProviderConfigUsage` that has reference to that `ProviderConfig`. However, there isn't anything that filters out other `ProviderConfigUsage`s which reference a different `ProviderConfig`. It seems this only does a check for an interface.

Reply:
I don't follow. We watch a particular GVK of `ProviderConfigUsage` - e.g. GCP PCUs - and enqueue a request for the `ProviderConfig` they reference below. 

https://github.com/crossplane/provider-gcp/pull/262/files#diff-37d5b8a93884a62afcabf3fc6dceeea0R44

Reply:
Hmm, I think you can achieve that by using `Owns(&v1beta1.ProviderConfigUsage{})`.

Reply:
>// Owns defines types of Objects being *generated* by the ControllerManagedBy, and configures the ControllerManagedBy to respond to
// create / delete / update events by *reconciling the owner object*.  This is the equivalent of calling
// Watches(&source.Kind{Type: <ForType-forInput>}, &handler.EnqueueRequestForOwner{OwnerType: apiType, IsController: true})

Oh, nevermind. PCs are not owner of PCUs.